hecuba_py/hecuba/hnumpy.py

```diff
@@ -107,14 +110,111 @@ def _store_meta(storage_args):
             raise ex
 
     @staticmethod
-    def load_array(storage_id, name):
-        hcache = StorageNumpy._create_hcache(storage_id, name)
-        result = hcache.get_numpy([storage_id])
-        if len(result) == 1:
-            return [result[0], hcache]
+    def reserve_numpy_array(storage_id, name):
+        '''Provides a numpy array with the number of elements obtained through storage_id'''
+
+        hcache = StorageNumpy._create_hcache(name)
+        result = hcache.allocate_numpy(storage_id)
+        if len(result) == 2:
+            return [result[0], result[1], hcache]
         else:
             raise KeyError
 
+    def generate_coordinates(self, coordinates):
+        if coordinates is None:
+            return []
+        # coords divided by number of elem in a row
+        coord = [coordinates[:, coord] // self._row_elem for coord in range(len(coordinates[0]))]
+        ranges = (range(*range_tuple) for range_tuple in zip(coord[0], coord[1] + 1))
+        keys = list(it.product(*ranges))
+        return keys
+
+    def format_coords(self, coord):
+        if not isinstance(coord, list):
```

Comment:
why that many formats?
I think you can force coord to be a slice or tuple of slices, by only modifying Line 137 and build a tuple of slices instead of a list